clients/drcachesim/tracer/instru_offline.cpp

```diff
@@ -403,17 +403,23 @@ int
 offline_instru_t::append_unit_header(byte *buf_ptr, thread_id_t tid, ptr_int_t window)
 {
     byte *new_buf = buf_ptr;
-    offline_entry_t *entry = (offline_entry_t *)new_buf;
-    entry->timestamp.type = OFFLINE_TYPE_TIMESTAMP;
-    uint64 frozen = frozen_timestamp_.load(std::memory_order_acquire);
-    entry->timestamp.usec = frozen != 0 ? frozen : instru_t::get_timestamp();
-    new_buf += sizeof(*entry);
+    new_buf += append_timestamp(new_buf);
```

Comment:
This timestamp is still emitted at the time we output the buffer right? Which means the timestamps inside the buffer may be smaller than the timestamp at the header?

Reply:
No, that's the old method: the new method since -align_endpoints is to write the timestamp at buffer creation time (that's why the post-syscall-buffer timestamp was actually a pre-syscall time, hence the misleading times).

Reply:
Oh right. So the timestamps should be in order already